api/src/main/java/org/apache/iceberg/actions/RewriteDataFilesV1.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.expressions.Expression;
+
+/**
+ * An action that rewrites data files.
+ */
+public interface RewriteDataFilesV1 extends SnapshotUpdate<RewriteDataFilesV1, RewriteDataFilesV1.Result> {
+  /**
+   * Pass a row filter to filter {@link DataFile}s to be rewritten.
+   * <p>
+   * Note that all files that may contain data matching the filter may be rewritten.
+   * <p>
+   * If not set, all files will be rewritten.
```

Comment:
Am I reading this correctly, that we rewrite all files whenever there's no filter? I thought that we only considered all files for rewrite, but left some alone under different criteria.

Could totally be nit-picking, but it makes it sound potentially much more expensive than it is if the docs say that every file is rewritten if some are only opened and then left as is. Might not be an important distinction though.

Reply:
Oh. I see that this is just a copy of the original one. Then not necessarily in scope to update the doc here.

Reply:
Seems I don't have permissions to close this. Something weird has happened to my GitHub account in this repo. On another PR I was tagged in, I'm not able to comment at all (even though it's still open). Feel free to close this!